akka-docs/src/main/paradox/scala/general/actor-systems.md

```diff
@@ -107,14 +107,153 @@ which is a single point of contention if over-used.
+but it will likely not exceed 50. Since many requests are being processed, the entire
+thread pool is starved. The blocking operations dominate threads in the dispatcher 
+such that it has no thread available to handle other requests.
+
+In essence, the `Thread.sleep` operation has dominated all threads and caused anything 
+executing on the default dispatcher to starve for resources (including any actor
+that you have not configured an explicit dispatcher for).
+
+## Solution: Dedicated dispatcher for blocking operations
+
+In `application.conf`, the dispatcher dedicated to blocking behaviour should
+be configured as follows:
+
+```conf
+my-blocking-dispatcher {
+  type = Dispatcher
+  executor = "thread-pool-executor"
+  thread-pool-executor {
+    // or in Akka 2.4.2+
+    fixed-pool-size = 16
+  }
+  throughput = 100
+}
+```
+
+There are many dispatcher options available which can be found in @ref[Dispatchers](../dispatchers.md).
+
+Here `thread-pool-executor` is used, which has a hardcoded limit of threads. It keeps a set number of threads
+available that allow for safe isolation of the blocking operations. The size settings should depend on the app's
+functionality and the number of cores the server has.
+
+Whenever blocking has to be done, use the above configured dispatcher
+instead of the default one:
+
+@@snip [BlockingDispatcherSample.scala](../../../../test/scala/docs/actor/BlockingDispatcherSample.scala) { #separate-dispatcher }
+
+The thread pool behaviour is shown in the figure.
+
+![dispatcher-behaviour-on-good-code.png](../../images/dispatcher-behaviour-on-good-code.png)
+
+the normal requests are easily handled by the default dispatcher - the
+green lines, which represent the actual execution.
+
+When blocking operations are issued, the `my-blocking-dispatcher`
+starts up to the number of configured threads. It handles sleeping. After
+a certain period of nothing happening to the threads, it shuts them down.
+
+If another bunch of operations have to be done, the pool will start new
+threads that will take care of putting them into sleep state, but the
+threads are not wasted.
+
+In this case, the throughput of other actors was not impacted -
+they were still served on the default dispatcher.
+
+This is the recommended way of dealing with any kind of blocking in reactive
+applications. It is referred to as "bulkheading" or "isolating" the bad behaving
+parts of an app. In this case, bad behaviour of blocking operations.
+
+There is a similar discussion in Akka HTTP docs section, @extref:[Handling blocking operations in Akka HTTP](akka.http:java/http/handling-blocking-operations-in-akka-http-routes.html#solution-dedicated-dispatcher-for-blocking-operations).
+
+## Available solutions to blocking operations
+
 The non-exhaustive list of adequate solutions to the “blocking problem”
-includes the following suggestions:
+includes the following suggestions, where the 3rd one was what we described in the previous section:
```

Comment:
Perhaps remove this but add in the 3rd point `(as explained in detail in section ....)`